crates/calibration/src/residuals.rs

```diff
@@ -1,21 +1,25 @@
-use linear_algebra::IntoTransform;
-use nalgebra::{DVector, Dyn, Owned, UnitQuaternion, Vector};
-use types::{camera_position::CameraPosition, field_dimensions::FieldDimensions};
+use color_eyre::Result;
+use nalgebra::{DVector, Dyn, Owned, Vector};
+use types::field_dimensions::FieldDimensions;
 
-use crate::{corrections::Corrections, lines::LinesError, measurement::Measurement};
+use crate::corrections::Corrections;
 
 pub type Residual = Vector<f32, Dyn, ResidualStorage>;
 pub type ResidualStorage = Owned<f32, Dyn>;
 
-pub fn calculate_residuals_from_parameters(
+pub fn calculate_residuals_from_parameters<MeasurementType, StructuredResidual>(
     parameters: &Corrections,
-    measurements: &[Measurement],
+    measurements: &[MeasurementType],
     field_dimensions: &FieldDimensions,
-) -> Option<Residual> {
+) -> Option<Residual>
+where
+    StructuredResidual: CalculateResiduals<MeasurementType>,
```

Comment:
1. Why is this type called "structured"?
2. Why is it called "residual" and not "residuals"?

Reply:
I'm not happy with the naming; reasoning to have the weird name, I do agree with you, it needs consistency.
Looked at a few literature and they use the term residuals for multiple values:

1. `Residual` -> nalgebra vector (current state of `main`) -> Should be `ResidualsVector` IMO
2. `StructuredResidual` -> What I wanted to mean was that this is a struct where calculation is done across values of a struct (i.e. different lines and angles of goal box) instead of doing a typical vector/ matrix style calculation with homogenous types (as in what a value represent*)

If you have some suggestion to replace `StructuredResidual`, it would be great, as I'm out of ideas :'(.
I thought of `ResidualProvider` or `ResidualImpl` but I don't like either of them very much!

Reply:
Maybe Tschadgippeddy can help here?

Reply:
I've currently converged at:

1. `ResidualVector` for the `nalgebra::vector`,
2. `ResidualsFromMeasurement` to replace `StructuredResidual`
3. As I mentioned in the chat, Preferably rename the `Measurement` structs to have prefixes so I can shorten `MeasurementType` to `Measurement`

Reply:
Done, let me know if this looks better :)

Tried "Tschadgippeddy" but it wasn't that good in giving a generic name. Maybe I need to study some 'prompt engineering'

Reply:
Fine for the moment 👍